packages/fetchai/connections/p2p_stub/connection.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains the p2p stub connection."""
+
+import logging
+import os
+import tempfile
+from pathlib import Path
+from typing import Union
+
+from aea.configurations.base import ConnectionConfig, PublicId
+from aea.connections.base import Connection
+from aea.connections.stub.connection import (
+    StubConnection,
+    _encode,
+    lock_file,
+)
+from aea.mail.base import Address, Envelope
+
+
+logger = logging.getLogger(__name__)
+
+
+class P2PStubConnection(StubConnection):
+    r"""A p2p stub connection.
+
+    This connection uses an existing directory as a Rendez-Vous point for agents to communicate locally.
+    Each connected agent will create a file named after its address/identity where it can receive messages.
+
+    The connection detects new messages by watchdogging the input file looking for new lines.
+    """
```

Comment:
```suggestion
    """
    A p2p stub connection.

    This connection uses an existing directory as a Rendez-Vous point for agents to communicate locally.
    Each connected agent will create a file named after its address/identity where it can receive messages.

    The connection detects new messages by watchdogging the input file looking for new lines.
    """
```

Reply:
For consistency